front/src/applications/stdcm/views/StdcmRequestModal.jsx

```diff
@@ -79,18 +79,16 @@ export default function StdcmRequestModal(props) {
               path_id: result.path.id,
             },
           }).then((simulationLocal) => {
-            const newSimulation = {};
-            newSimulation.trains = [...simulationLocal];
-            newSimulation.trains = [...newSimulation.trains, fakedNewTrain];
+            const trains = [...simulationLocal, fakedNewTrain];
             const consolidatedSimulation = createTrain(
               dispatch,
               KEY_VALUES_FOR_CONSOLIDATED_SIMULATION,
-              newSimulation.trains,
+              trains,
               t
             );
             dispatch(updateConsolidatedSimulation(consolidatedSimulation));
-            dispatch(updateSimulation(newSimulation));
-            dispatch(updateSelectedTrain(newSimulation.trains.length - 1));
+            dispatch(updateSimulation({ trains }));
+            dispatch(updateSelectedTrainId(trains[0].id));
```

Comment:
This is for STDCM projection :)
```suggestion
            dispatch(updateSelectedTrainId(fakeNewTrain.id));
```

Reply:
done ! Nicely spotted :)